driver-core/src/main/com/mongodb/client/model/expressions/MqlExpression.java

```diff
@@ -118,4 +119,36 @@ public <R extends Expression> R cond(final R left, final R right) {
         return newMqlExpression(ast("$cond", left, right));
     }
 
+
+    /** @see ArrayExpression */
+
+    @Override
+    public <R extends Expression> ArrayExpression<R> map(final Function<T, ? extends R> in) {
+        T varThis = variable("$$this");
+        return new MqlExpression<>((cr) -> astDoc("$map", new BsonDocument()
+                .append("input", this.toBsonValue(cr))
+                .append("in", extractBsonValue(cr, in.apply(varThis)))
+                .toBsonDocument(BsonDocument.class, cr)).apply(cr));
+    }
+
+    @Override
+    public ArrayExpression<T> filter(final Function<T, BooleanExpression> cond) {
```

Comment:
I think we should implement this parameter in something like the following way instead:

`array.filter(condition).limit(10)`

Reply:
What's the plan for this functionality? Given that this PR implements `$filter`, I'd expect the functionality added in this PR.

Reply:
@stIncMale MQL has added this parameter for the sake of convenience. This convenience helps programmers writing MQL (with query size), and possibly the server in interpreting MQL. However, it is detrimental for users of our API, because it unnecessarily introduces two ways of doing things, and the mechanism that it introduces is inelegant and bloats our API. So I do not think it should be implemented.

@jyemin I believe that method that would correspond to limit is [$firstN](https://www.mongodb.com/docs/manual/reference/operator/aggregation/firstN-array-element/). I do not like that the names are inconsistent, and would like to address that, but do not see a clean way to do so.